### PR TITLE
Check for null and empty scopes in managed identity

### LINKED ISSUE #4332: [Bug] Managed identity crashes when resource is null
### Library version used

4.54.1

### .NET version

Any.

### Scenario

ManagedIdentityClient - managed identity

### Is this a new or an existing app?

None

### Issue description and reproduction steps

Managed identity code crashes when [default suffix is removed from the resource](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/1b56a35f5add14c9bb439d50b3ae95368f5c107d/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs#L43) if the resource is null.

### Relevant code snippets

_No response_

### Expected behavior

_No response_

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

_No response_

### Solution and workarounds

_No response_

### PR DESCRIPTION
**Pull request by gladjohn**
Fixes #4332.
Check for null and empty scopes in managed identity

### CODE REVIEW

**Review by trwalke — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs [hunk 1]
```diff
@@ -45,6 +45,13 @@ public sealed class ManagedIdentityApplication
         /// <inheritdoc/>
         public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIdentity(string resource)
         {
+            if (string.IsNullOrEmpty(resource))
+            {
+                throw new MsalClientException(
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs:50]: Throw ArgumentNullException instead. We shouldn't invent exceptions for this.

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
NPE is fine. We should not throw MSAL Client Ex for simple arg validation

**Comment by gladjohn** [comment]
> ArgNullEx is better here. We should not throw MSAL Client Ex for simple arg validation

updated

File: src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs (hunk 1 above)
Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs]: I agree that Arg Ex is better, but this is breaking change? (Although I'd consider throwing an MsalEx a bug...)

File: tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs [hunk 2]
```diff
@@ -300,6 +300,29 @@ await mi.AcquireTokenForManagedIdentity(resource)
...
+                var miBuilder = ManagedIdentityApplicationBuilder.Create(ManagedIdentityId.SystemAssigned)
+                    .WithHttpManager(httpManager);
+
+                var mi = miBuilder.Build();
+
+                ArgumentNullException ex = await Assert.ThrowsExceptionAsync<ArgumentNullException>(async () =>
+                    await mi.AcquireTokenForManagedIdentity(resource)
+                    .ExecuteAsync().ConfigureAwait(false)).ConfigureAwait(false);
+
+                Assert.IsNotNull(ex);
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs:322]: nit: not needed. The Assert above will throw if error is null.

**Review by pmaytak — APPROVED** [review]

**Review by bgavrilMS — APPROVED** [review]

**Comment by pmaytak** [comment]
@gladjohn What does moving the null check accomplish? Is null resource not caught right now?

**Comment by gladjohn** [comment]
> @gladjohn What does moving the null check accomplish? Is null resource not caught right now?

For MI, we need to [remove the default suffix](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/1b56a35f5add14c9bb439d50b3ae95368f5c107d/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs#L43) if present. This is done before the null check. So moving the null check will not cause this to crash.

**Review by neha-bhargava — APPROVED** [review]

**Pull request merged**